src/data/object.lisp

```diff
@@ -153,14 +153,15 @@ reads will return nil."))
                (format stream "The transient slot ~A in class ~A was defined as having an initarg, which is not supported"
                        slot-name (class-name class))))))
 
-(defmethod direct-slot-definition-class ((class persistent-class) &key initargs transient name)
-  ;; It might be better to do the error checking in an
-  ;; initialize-instance method of persistent-direct-slot-definition
-  (when (and initargs transient)
-    (error 'transient-slot-cannot-have-initarg :class class :slot-name name))
-  'persistent-direct-slot-definition)
-
-(defmethod effective-slot-definition-class ((class persistent-class) &key)
+(eval-when (:compile-toplevel :load-toplevel :execute)
```

Comment:
Without this, LW fails to compile this file:
```
**++++ Error in (DEFCLASS BKNR.DATASTORE:STORE-OBJECT):                                                                                                                                                            
  #<COMMON-LISP:STANDARD-GENERIC-FUNCTION DIRECT-SLOT-DEFINITION-CLASS 41A02E2CFC> is called with keyword :NAME among the arguments (#<PERSISTENT-CLASS STORE-OBJECT 40201ED1B3> :NAME ID :READERS (STORE-OBJECT-I\
D) :INITARGS (:ID) :TYPE INTEGER :INDEX-MAPVALUES MAP-STORE-OBJECTS :INDEX-VALUES ALL-STORE-OBJECTS :INDEX-READER STORE-OBJECT-WITH-ID :INDEX-INITARGS (:TEST (FUNCTION EQL)) :INDEX-TYPE UNIQUE-INDEX) which is n\
ot one of (:INDEX-TYPE :INDEX).                                                                                                                                                                                    
....                                                                                                                                                                                                               
; *** 1 error detected, no fasl file produced.                                                                                                                                                                     

```

Surprisingly this compiles when I load it in the LW IDE, but fails when I load it from Emacs in a non IDE instance. So I'm not sure what's happening. LW doesn't give me a stack trace at this point to figure out why this failed.